packages/knip/src/plugins/eleventy/helpers.ts

```diff
@@ -0,0 +1,153 @@
+  setNunjucksPrecompiledTemplates() {}
+  setEjsOptions() {}
+  setDynamicPermalinks() {}
+  setUseGitIgnore() {}
+  addShortcode() {}
+  addAsyncShortcode() {}
+  addNunjucksAsyncShortcode() {}
+  addNunjucksShortcode() {}
+  addLiquidShortcode() {}
+  addHandlebarsShortcode() {}
+  addPairedShortcode() {}
+  addPairedAsyncShortcode() {}
+  addPairedNunjucksAsyncShortcode() {}
+  addPairedNunjucksShortcode() {}
+  addPairedLiquidShortcode() {}
+  addPairedHandlebarsShortcode() {}
+  addJavaScriptFunction() {}
+  setDataDeepMerge() {}
+  isDataDeepMergeModified() {}
+  addWatchTarget() {}
+  setWatchJavaScriptDependencies() {}
+  setServerOptions() {}
+  setBrowserSyncConfig() {}
+  setChokidarConfig() {}
+  setWatchThrottleWaitTime() {}
+  setFrontMatterParsingOptions() {}
+  setQuietMode() {}
+  addExtension() {}
+  addDataExtension() {}
+  setUseTemplateCache() {}
+  setPrecompiledCollections() {}
+  setServerPassthroughCopyBehavior() {}
+  addUrlTransform() {}
+  setDataFileSuffixes() {}
+  setDataFileBaseName() {}
+  getMergingConfigObject() {}
+
+  _uniqueId = {};
+  events = {};
+  benchmarkManager = {};
+  benchmarks = {};
+  collections = {};
+  precompiledCollections = {};
+  templateFormats = {};
+  liquidOptions = {};
+  liquidTags = {};
+  liquidFilters = {};
+  liquidShortcodes = {};
+  liquidPairedShortcodes = {};
+  nunjucksEnvironmentOptions = {};
+  nunjucksPrecompiledTemplates = {};
+  nunjucksFilters = {};
+  nunjucksAsyncFilters = {};
+  nunjucksTags = {};
+  nunjucksGlobals = {};
+  nunjucksShortcodes = {};
+  nunjucksAsyncShortcodes = {};
+  nunjucksPairedShortcodes = {};
+  nunjucksAsyncPairedShortcodes = {};
+  javascriptFunctions = {};
+  markdownHighlighter = null;
+  libraryOverrides = {};
+  passthroughCopies = {};
+  layoutAliases = {};
+  layoutResolution = true;
+  linters = {};
+  transforms = {};
+  activeNamespace = '';
+  DateTime = {};
+  dynamicPermalinks = true;
+  useGitIgnore = true;
+  ignores = new Set();
+  watchIgnores = new Set();
+  dataDeepMerge = true;
+  extensionMap = new Set();
+  watchJavaScriptDependencies = true;
+  additionalWatchTargets = [];
+  serverOptions = {};
+  globalData = {};
+  chokidarConfig = {};
+  watchThrottleWaitTime = 0;
+  dataExtensions = new Map();
+  quietMode = false;
+  plugins = [];
+  _pluginExecution = false;
+  useTemplateCache = true;
+  dataFilterSelectors = new Set();
+  libraryAmendments = {};
+  serverPassthroughCopyBehavior = '';
+  urlTransforms = [];
+  dataFileSuffixesOverride = false;
+  dataFileDirBaseNameOverride = false;
+  frontMatterParsingOptions = {
+    engines: {},
+  };
+  templateFormatsAdded = {};
+}
+
+// https://www.11ty.dev/docs/config/#configuration-options
+export const defaultEleventyConfig: EleventyConfig = {
+  dir: {
+    input: '.',
+    output: '_site',
+    includes: '_includes',
+    layouts: '_includes',
+    data: '_data',
+  },
+  templateFormats: 'html,liquid,ejs,md,hbs,mustache,haml,pug,njk,11ty.js',
```

Comment:
This is just the default value for this, I probably should remove all the other non-JS ones?

Reply:
Yes, would be good to remove indeed. They would not actually get added eventually (there's a filter for unknown extensions), but it's just a waste.